dbms/src/Flash/Coprocessor/InterpreterUtils.cpp

```diff
@@ -162,6 +166,35 @@ void orderStreams(
     }
 }
 
+void executeLocalSort(
+    PipelineExecutorStatus & exec_status,
+    PipelineExecGroupBuilder & group_builder,
+    const SortDescription & order_descr,
+    size_t limit,
+    const Context & context,
+    const LoggerPtr & log)
+{
+    auto input_header = group_builder.getCurrentHeader();
+    if (SortHelper::isSortByConst(input_header, order_descr))
+    {
+        // For order by const and limit != 0, we will generate LimitOperator directly.
+        if (limit != 0)
+        {
+            auto global_limit = std::make_shared<GlobalLimitTransformAction>(input_header, limit);
```

Comment:
Why use `global_limit` when the sort is local sort?

Reply:
ok, use `local_limit` instead